cmd/compute-domain-kubelet-plugin/computedomain.go

```diff
@@ -325,7 +325,7 @@ func (m *ComputeDomainManager) GetNodeIPs(ctx context.Context, cdUID string) ([]
 	}
 
 	if cd.Status.Nodes == nil {
-		return nil, fmt.Errorf("error getting status of nodes in ComputeDomain: %w", err)
+		return nil, fmt.Errorf("error getting status of nodes in ComputeDomain: no nodes set yet")
```

Comment:
Not critical, but should this just be:
```suggestion
		return nil, fmt.Errorf("no nodes set")
```
(I would remove the `yet` personally)

Going from `Nodes == nil` to error getting status of nodes implies intimate knowledge of `GetComputDomain`. Would it be better to just return an error from there instead?

We could also remove this check entirely since`len(nil) == 0` it should fail the check bellow as well (assuming that `NumNodes == 0` is not a valid configuration).

Reply:
Evan! Thanks for looking :)

> ... implies intimate knowledge of GetComputDomain. Would it be better to just return an error from there instead?

Good question. I'd say "no" because `GetComputeDomain()` is expected to return the CD object regardless of where the CD is in its lifecycle.

Hence, it can at times return a CD that does not have `.Status.Nodes` populated yet. That's an expected outcome IIUC.

Node status is added out-of-band via (potentially many calls from different nodes to) `AddNodeStatusToComputeDomain()` at an unpredictable point in time.

That is, via multiple calls to `GetNodeIPs()` we detect and respond to changes in a CD's lifecycle. Only within `GetNodeIPs()` it is an error when no node status has been set yet.

> should this just be: ... `no nodes set`

I like to shorten this further, yes! In terms of concatenation of error messages, let's optimize for readability in case where this is prefixed with "error getting node IPs: ..."

I like "error getting node IPs: no nodes set yet for ComputeDomain"

> I would remove the yet personally

The "yet" does not convey critical information, agree. It does however vaguely communicate the notion of an expected, temporary error (as opposed to unexpected, permanent error -- a distinction that is best done not just with words like "yet" :-)).

> We could also remove this check entirely sincelen(nil) == 0

Ack. But I think it's good to have the fine-grained output so that we can see state transitioning explicitly: no nodes seen -> not all nodes seen -> all nodes seen

Reply:
OK. I think having different error messages for the two cases is fine. On the use of "yet". Looking at the logic, it is not within the scope of this function to determine whether the error is treated as transient or not. I would then remove the "yet" and have intent to retry conveyed in the caller or through some other mechanism.

```
"error getting node IPs: no nodes set for ComputeDomain"
```
sounds good. (I won't fight hard on the `"yet"` :) )

Reply:
One thing to note is that the `error getting node IPs` would be prepended in the caller, so here we would only have:
```suggestion
		return nil, fmt.Errorf("no nodes set for ComputeDomain")
```

Reply:
Done yet? Not yet, we haven't yet discussed:

> not all nodes populated in ComputeDomain status yet

That's the current error message right below the one we're discussing here.

Reply:
Pushed commit, without the "yet" :).